### PR TITLE
Add detailed schema for tidb cluster configurations

### LINKED ISSUE #1308: Generate detailed openapi schema for configurations

### PR DESCRIPTION
**Pull request by aylei**
Signed-off-by: Aylei <rayingecho@gmail.com>

close #1308 
### What problem does this PR solve? 
This PR add a validation schema for TiDB configurations.

Prior to `v1.15`, kubernetes do not have [mature support for CRD schema](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#specifying-a-structural-schema), in our scenario, kube-apiserver do not publish the schema of CRD and CRD do not allow specifying properties for fields like `default`, `maximum`. So the current work around is generate one and provide it via a go interface.

This PR adjust the struct of config in favor of schema generation, `default` value and `validation` of fields will be added in #1309 

UPDATE: discussed with @shuijing198799 , it is better to generate schema in client.

@weekface @LinuxGit @tennix @shuijing198799 PTAL

### CODE REVIEW

File: pkg/schema/types.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// +groupName=schema.pingcap.com
+// +versionName=v1alpha1
+package schema
+
+import (
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:resource
```
Comment (aylei) [inline pkg/schema/types.go]: `controller-gen` do not support filtering resources in the same package, create a dummy type to generate the configuration schema.

File: pkg/apis/pingcap/v1alpha1/pd_config.go [hunk 2]
```diff
@@ -261,10 +239,10 @@ type PDScheduleConfig struct {
 	MaxMergeRegionKeys *uint64 `toml:"max-merge-region-keys,omitempty,omitempty" json:"max-merge-region-keys,omitempty"`
 	// SplitMergeInterval is the minimum interval time to permit merge after split.
 	// +optional
-	SplitMergeInterval typeutil.Duration `toml:"split-merge-interval,omitempty,omitempty" json:"split-merge-interval,omitempty"`
+	SplitMergeInterval string `toml:"split-merge-interval,omitempty,omitempty" json:"split-merge-interval,omitempty"`
```
Comment (tennix) [inline pkg/apis/pingcap/v1alpha1/pd_config.go]: Two `omitempty` for toml?

Comment (aylei) [reply pkg/apis/pingcap/v1alpha1/pd_config.go]: Good cach

**Comment by aylei** [comment]
@tennix @shuijing198799 @LinuxGit  Addressed, PTAL again

File: pkg/apis/pingcap/v1alpha1/pd_config.go (hunk 2 above)
Comment (shuijing198799) [reply pkg/apis/pingcap/v1alpha1/pd_config.go]: catch ？ hahahahhahahahah

**Pull request merged**